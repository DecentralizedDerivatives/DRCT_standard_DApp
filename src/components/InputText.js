import React from 'react';

export const InputText = ({ label, input, meta: { touched, error } }) => {
  return (
    <div className="mv4 w-100">
      <div className="b sans-serif pv2 w-100">{label}</div>
      <input
        {...input}
        placeholder={label}
        type="text"
        className="pa2 ba b--black-40 w-100"
      />

      {touched && (error && <span>{error}</span>)}
    </div>
  );
};

export default InputText;
