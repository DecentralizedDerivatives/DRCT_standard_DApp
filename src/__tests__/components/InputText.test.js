import InputText from '../../components/InputText';

describe('<InputText />', () => {
  it('renders the component', () => {
    const wrapper = shallow(<InputText store={initStore()} meta={{}} />);
    expect(wrapper).toMatchSnapshot();
  });
});
