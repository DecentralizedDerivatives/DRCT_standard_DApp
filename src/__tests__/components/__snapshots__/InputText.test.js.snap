// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputText /> renders the component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <InputText
    meta={Object {}}
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <label
          className="b sans-serif pv2 w-100"
        />,
        <input
          type="text"
        />,
        undefined,
      ],
      "className": "input-text",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "b sans-serif pv2 w-100",
        },
        "ref": null,
        "rendered": null,
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "placeholder": undefined,
          "type": "text",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      undefined,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <label
            className="b sans-serif pv2 w-100"
          />,
          <input
            type="text"
          />,
          undefined,
        ],
        "className": "input-text",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "b sans-serif pv2 w-100",
          },
          "ref": null,
          "rendered": null,
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "placeholder": undefined,
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        undefined,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
